app/src/main/java/chat/rocket/android/settings/presentation/settingPresenter.kt

```diff
@@ -0,0 +1,62 @@
+package chat.rocket.android.settings.presentation
+
+import chat.rocket.android.core.lifecycle.CancelStrategy
+import chat.rocket.android.server.domain.GetCurrentServerInteractor
+import chat.rocket.android.server.infraestructure.RocketChatClientFactory
+import chat.rocket.android.util.extension.launchUI
+import chat.rocket.android.util.retryIO
+import chat.rocket.common.util.ifNull
+import chat.rocket.core.RocketChatClient
+import chat.rocket.core.internal.rest.me
+import chat.rocket.core.internal.rest.requestDataDownload
+import kotlinx.coroutines.experimental.DefaultDispatcher
+import kotlinx.coroutines.experimental.withContext
+import javax.inject.Inject
+
+class settingPresenter @Inject constructor(
+    private val view: SettingsView,
+    private val strategy: CancelStrategy,
+    serverInteractor: GetCurrentServerInteractor,
+    factory: RocketChatClientFactory
+
+
+) {
+    private val serverUrl = serverInteractor.get()!!
+    private val client: RocketChatClient = factory.create(serverUrl)
+
+
+
+
+
+    fun downloadData() {
+        launchUI(strategy) {
+            //view.showLoading()
+            try {
+                withContext(DefaultDispatcher) {
+                    // REMARK: Backend API is only working with a lowercase hash.
+                    // https://github.com/RocketChat/Rocket.Chat/issues/12573
+
+                    val me = retryIO("me") { client.me() }
```

Comment:
You don't need to invoke the `client.me()` to get the user id. You are using more background task doing that here.